src/main.ts

```diff
@@ -0,0 +1,656 @@
+
+  return c.text(summary);
+});
+
+app.get("PRs", async (c) => {
+  const userID = c.req.query("user_id");
+
+  if (!userID) {
+    return c.json({ message: "unauthorized" }, { status: 401 });
+  }
+
+  const rows = (await db
+    .query("select * from tokens where user_id = $userID")
+    .all({ $userID: userID as string })) as TokenRow[];
+  const Client = getGitHubClient();
+
+  if (rows.length === 0) {
+    throw new Error("User not found in database. Log in with github first");
+  }
+
+  const { refresh_token, expiration_date } = rows[0];
+
+  // if present get accessToken and its expiration date from memory
+  let accessToken: string | undefined;
+  let expirationAccessToken: string = new Date().toISOString();
+  const token = map.get(userID);
+  if (token && token.expiresEpoch > Date.now()) {
+    accessToken = token.accessToken;
+    expirationAccessToken = new Date(token.expiresEpoch).toISOString();
+  }
+
+  // decrypt the refresh token
+  const buffer = Buffer.from(refresh_token, "base64");
+  const uint8Array = new Uint8Array(buffer);
+  const decryptedTemp = await decrypt(uint8Array, key);
+  const decrypted = new TextDecoder().decode(decryptedTemp);
+
+  // create github client
+  const client = new Client({
+    authStrategy: createOAuthUserAuth,
+    auth: {
+      clientType: "github-app",
+      clientId: config.oauth.github.clientID,
+      clientSecret: config.oauth.github.secret,
+      token: accessToken || "",
+      refreshToken: decrypted,
+      expiresAt: expirationAccessToken,
+      refreshTokenExpiresAt: new Date(expiration_date).toISOString(),
+    } satisfies GitHubAppStrategyOptionsExistingAuthenticationWithExpiration,
+  });
+
+  const [start, end] = await getDates();
+
+  // NEEDS TO BE SET IN PHILL
+  const repoName = "alumni-api";
+  const message: ListMessage = {
+    repos: repoName,
+    since: start,
+    until: end,
+    author: rows[0].user_name,
+  };
+
+  const listPrs = await listPullRequests(client, message);
+
+  return c.text(await summarizePrs(listPrs, repoName));
+});
+
+app.get("summary", async (c) => {
+  const userID = c.req.query("user_id");
+
+  if (!userID) {
+    return c.json({ message: "unauthorized" }, { status: 401 });
+  }
+
+  const rows = (await db
+    .query("select * from tokens where user_id = $userID")
+    .all({ $userID: userID as string })) as TokenRow[];
+  const Client = getGitHubClient();
+
+  if (rows.length === 0) {
+    throw new Error("User not found in database. Log in with github first");
+  }
+  const { refresh_token, expiration_date } = rows[0];
+
+  if (new Date(rows[0].expiration_date) < new Date()) {
+    throw "refresh token has expired log in again";
+  }
+
+  // decrypt the refresh token
+  const buffer = Buffer.from(refresh_token, "base64");
+  const uint8Array = new Uint8Array(buffer);
+  const decryptedTemp = await decrypt(uint8Array, key);
+  const decrypted = new TextDecoder().decode(decryptedTemp);
```

Comment:
You do the same thing in 3 handlers, I think you can extract it to `decodeRefreshToken` or something similar

Reply:
For phill I would only be using the summary of both the prs and the commits so then it would only be used once. That is why I chose not to replace it with an external function.